src/bin/cargo-flamegraph.rs

```diff
@@ -320,19 +322,46 @@ fn workload(
     result
 }
 
-fn find_unique_bin_target() -> String {
+#[derive(Clone, Debug)]
+struct BinaryTarget {
+    package: String,
+    target: String,
+}
+
+impl std::fmt::Display for BinaryTarget {
+    fn fmt(
+        &self,
+        f: &mut std::fmt::Formatter,
+    ) -> std::fmt::Result {
+        write!(
+            f,
+            "target {} in package {}",
+            self.target, self.package
+        )
+    }
+}
+
+fn find_unique_bin_target() -> BinaryTarget {
     let mut metadata_command = MetadataCommand::new();
     metadata_command.no_deps();
     let metadata = metadata_command
         .exec()
         .expect("failed to access crate metadata");
 
-    let bin_targets: Vec<String> = metadata
+    let bin_targets: Vec<BinaryTarget> = metadata
         .packages
         .into_iter()
-        .flat_map(|p| p.targets.into_iter())
-        .filter(|t| t.kind.contains(&"bin".into()))
-        .map(|t| t.name)
+        .flat_map(|p| {
+            let Package { targets, name, .. } = p;
+            targets
```

Comment:
Why can't we use `p.targets` directly anymore, as before?

Reply:
No because `p` is used in the closure again and the borrow checker does not like that:
```
error[E0382]: use of partially moved value: `p`
   --> src/bin/cargo-flamegraph.rs:356:22
    |
355 |                 .into_iter()
    |                  ----------- `p.targets` partially moved due to this method call
356 |                 .map(move |t| (p.name.clone(), t))
    |                      ^^^^^^^^  ------ use occurs due to use in closure
    |                      |
    |                      value used here after partial move
    |
note: this function takes ownership of the receiver `self`, which moves `p.targets`
```